scripts/pi-hole/php/add.php

```diff
@@ -8,100 +8,66 @@
 
 require_once('auth.php');
 
-$type = $_POST['list'];
+$list = $_POST['list'];
 
 // Perform all of the authentication for list editing
 // when NOT invoked and authenticated from API
 if (empty($api)) {
-    list_verify($type);
+    list_verify($list);
 }
 
-// Don't check if the added item is a valid domain for regex expressions. Regex
-// filters are validated by FTL on import and skipped if invalid
+// Only check domains we add to the exact lists.
+// Regex are validated by FTL during import
 $check_lists = ["white","black","audit"];
 if(in_array($list, $check_lists)) {
     check_domain();
 }
 
-// Escape shell metacharacters
-$domains = explode(",",$_POST['domain']);
+// Split individual domains into array
+$domains = explode(" ",$_POST['domain']);
```

Comment:
Users may have multiple spaces between domains, which would cause this code to add empty domains to the list. Try this: https://stackoverflow.com/a/1792977